core/src/test/java/org/bitcoinj/core/strategies/DenseThenSparseStrategyTest.java

```diff
@@ -0,0 +1,50 @@
+package org.bitcoinj.core.strategies;
+
+import org.bitcoinj.core.Block;
+import org.bitcoinj.core.BlockChain;
+import org.bitcoinj.core.Utils;
+import org.bitcoinj.params.UnitTestParams;
+import org.bitcoinj.store.BlockStore;
+import org.bitcoinj.store.MemoryBlockStore;
+import org.bitcoinj.wallet.Wallet;
+import org.junit.Before;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.bitcoinj.testing.FakeTxBuilder.createFakeBlock;
+
+public class DenseThenSparseStrategyTest {
+
+    private static final Logger log = LoggerFactory.getLogger(DenseThenSparseStrategyTest.class);
+    private DenseThenSparseStrategy denseThenSparseStrategy;
+    private BlockChain blockChain;
+
+    @Before
+    public void setUp() throws Exception {
+        denseThenSparseStrategy = new DenseThenSparseStrategy();
+        denseThenSparseStrategy.setNetworkParameters(UnitTestParams.get());
+        BlockStore blockStore = new MemoryBlockStore(UnitTestParams.get());
+        Wallet wallet = new Wallet(UnitTestParams.get());
+        blockChain = new BlockChain(UnitTestParams.get(), wallet,blockStore);
+
+        Block b1 = createFakeBlock(blockStore, Block.BLOCK_HEIGHT_GENESIS).block;
+        blockChain.add(b1);
+
+        for (int i=blockChain.getBestChainHeight() + 1; i<100; i++) {
+            try {
+                Block b = createFakeBlock(blockStore, i).block;
+                blockChain.add(b);
+            } catch (Exception e) {
+                log.error("fails at i {}", i, e);
```

Comment:
The test set-up does not complete. What am I doing wrong?

SEVERE: fails at i 10
org.bitcoinj.core.VerificationException: Could not verify block:
Caused by: org.bitcoinj.core.VerificationException: Network provided difficulty bits do not match what was calculated: 201fffff vs 207fffff
	at org.bitcoinj.params.AbstractBitcoinNetParams.checkDifficultyTransitions(AbstractBitcoinNetParams.java:135)
	at org.bitcoinj.core.AbstractBlockChain.add(AbstractBlockChain.java:488)